lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -81,9 +81,11 @@ def problem_scores(self):
         ):
             block = self.course_data.structure[block_key]
             if getattr(block, 'has_score', False):
-                locations[block_key] = get_score(
+                problem_score = get_score(
                     submissions_scores={}, csm_scores={}, persisted_block=None, block=block,
                 )
+                if problem_score:
```

Comment:
If this is meant to guard against None, would `if problem_score is not None:` be better?  Or do you also want to skip the next line if problem_score is zero?